inference-engine/tests/functional/shared_test_classes/src/single_layer/activation.cpp

```diff
@@ -148,6 +148,13 @@ InferenceEngine::Blob::Ptr ActivationLayerTest::GenerateInput(const InferenceEng
             data_start_from = 0;
         }
     }
+    // To avoid extremely small input values which lose sign during quantization
+    if (activationType == ngraph::helpers::ActivationTypes::Sign
+        && targetDevice == CommonTestUtils::DEVICE_GNA
+        && (info.getPrecision() == InferenceEngine::Precision::FP16 ||
+            info.getPrecision() == InferenceEngine::Precision::FP32)) {
+        resolution = 3072;
```

Comment:
Is it possible to handle it not in common part?

Reply:
@nkogteva, It is possible, but I'm afraid that such a solution will contain code duplication.